core/src/main/java/io/onetable/client/TableFormatClientFactory.java

```diff
@@ -43,15 +38,21 @@ public static TableFormatClientFactory getInstance() {
 
   public TargetClient createForFormat(
       String tableFormat, PerTableConfig perTableConfig, Configuration configuration) {
-    switch (tableFormat) {
-      case ICEBERG:
-        return new IcebergClient(perTableConfig, configuration);
-      case DELTA:
-        return new DeltaClient(perTableConfig, configuration);
-      case HUDI:
-        return new HudiTargetClient(perTableConfig, configuration);
-      default:
-        throw new NotSupportedException("Target format is not yet supported: " + tableFormat);
+    TargetClient targetClient = createTargetClientForName(tableFormat);
+    if (targetClient == null) {
+      throw new NotSupportedException("Target format is not yet supported: " + tableFormat);
     }
+    targetClient.init(perTableConfig, configuration);
+    return targetClient;
+  }
+
+  public TargetClient createTargetClientForName(String tableFormatName) {
```

Comment:
nit: This method does not require public access.

Reply:
This depends on the programming model going forward. This allows for application code to create a targetClient then set values and call init(). If we want to completely hide that going forward then we can make this private. I figured we would let that be decided as a follow up but can change it now and we can always open it up later.

Reply:
If this method is meant for public use, please provide a javadoc comment that explains its purpose.

Reply:
I guess that I'll add one to the existing method as well...

Reply:
done.